### PR TITLE
Fixes Property links don't work in de-formal language

### PR DESCRIPTION
**Pull request by thomas-topway-it**
fixes https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5342

However before pulling the edits, it should be further tested and the following issue should be addressed:

- to what extent the tag de-x-formal is used in MW and SMW and why de-formal isn't completely replaced with de-x-formal in i18n files ? (or on the contrary de-formal inserted in the language-subtag-registry ? https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry

otherwise, if we keep the mapping, we have to ensure that every time the language code is handled, it expects either 'de-formal' or de-x-formal' (the present pull request do it within the scope of SMW, but not of MW)

Also refer to the following:

- https://schneegans.de/lv/?tags=de-x-formal&Format=text
- https://doc.wikimedia.org/mediawiki-core/1.33.0/php/LanguageCode_8php_source.html
- https://phabricator.wikimedia.org/T125073
- https://datatracker.ietf.org/doc/html/rfc5646 (Private use subtag)

### CODE REVIEW

**Comment by thomas-topway-it** [comment]
@gesinn-it-gea and @kghbln do you want to take a look ?

File: src/DataValues/MonolingualTextValue.php [hunk 1]
```diff
@@ -277,7 +289,12 @@ public function getTextValueByLanguageCode( $languageCode ) {
 			return null;
 		}
 
-		if ( $list['_LCODE'] !== Localizer::asBCP47FormattedLanguageCode( $languageCode ) ) {
+		$mappedLanguageCode = !array_key_exists( $list['_LCODE'], $this->nonstandardLanguageCodeMapping ) ? $list['_LCODE']
```
Comment (Nikerabbit) [inline src/DataValues/MonolingualTextValue.php:292]: More idiomatic way to write this is `$mappedLanguageCode = $this->nonstandardLanguageCodeMapping[$list['_LCODE']] ?? $list['_LCODE']`

File: src/DataValues/LanguageCodeValue.php [hunk 2]
```diff
@@ -49,15 +64,22 @@ protected function parseUserValue( $userValue ) {
 
 		// Checks whether the language tag is valid in MediaWiki for when
 		// it is not executed in a query context
-		if ( !$this->getOption( self::OPT_QUERY_CONTEXT ) && !Localizer::isKnownLanguageTag( $languageCode ) ) {
+		
+		// ensure non-standard language codes are mapped to
+		// their canonical form (e.g. de-x-formal to de-formal)
+
+		$mappedLanguageCode = !in_array( $languageCode, $this->nonstandardLanguageCodeMapping ) ? $languageCode
```
Comment (Nikerabbit) [inline src/DataValues/LanguageCodeValue.php:71]: `in_array` isn't needed. `array_search( $languageCode, $this->nonstandardLanguageCodeMapping ) ?: $languageCode` does the same thing.

Comment (Nikerabbit) [reply src/DataValues/LanguageCodeValue.php:71]: Yes bcp47 should be fine to use when formatting language codes for display. The list of deprecated language codes is changing over time, so it's possible the behavior is not the same across all MediaWiki versions.

**Comment by jaideraf** [comment]
Due to https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/5530 merge, this one can be closed, right?

**Comment by JeroenDeDauw** [comment]
> Due to #5530 merge, this one can be closed, right?

Dunno, @krabina reported the underlying issue (https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5342) was not fixed. So perhaps we should revert #5530.

**Pull request closed without merging**